external-crates/move/crates/move-compiler/tests/move_check/parser/friend_decl_inside_function.exp

```diff
@@ -8,8 +8,11 @@ error[E01002]: unexpected token
   │         Expected an expression term
 
 error[E01002]: unexpected token
-  ┌─ tests/move_check/parser/friend_decl_inside_function.move:9:1
+  ┌─ tests/move_check/parser/friend_decl_inside_function.move:6:16
   │
-9 │ }
-  │ ^ Invalid code unit. Expected 'address' or 'module'. Got '}'
+6 │         friend 0x42::A;
```

Comment:
This reset seems too hard. Shouldn't we have an exp follow set? 

Alternatively, we could just parse the `friend` decl anyway here (as a special case) and discard it.